### PR TITLE
implement any kernel

### PR DESCRIPTION
**Pull request by ritchie46**
First part for #738. I'll continue with the `all` after first review.

### CODE REVIEW

File: src/compute/boolean.rs [hunk 1]
```diff
@@ -185,3 +185,39 @@ pub fn or_scalar(array: &BooleanArray, scalar: &BooleanScalar) -> BooleanArray {
...
 }
+
+/// Check if any of the values in the array is `true`
+pub fn any(array: &BooleanArray) -> bool {
+    if array.is_empty() {
+        false
+    } else if array.validity().is_some() {
+        array.into_iter().any(|v| v == Some(true))
+    } else {
+        let vals = array.values();
```
Comment (jorgecarleitao) [inline src/compute/boolean.rs:196]: Here, we need to compute if any entry is set. isn't that equivalent to `array.values().null_count() > 0`?

Comment (ritchie46) [reply src/compute/boolean.rs:196]: Indeed. But this function returns on first `true`  value that is found, whereas `null_count` needs to scan the whole array.

Comment (jorgecarleitao) [reply src/compute/boolean.rs:196]: `null_count` is cached (yes, we also cache it for the values of the `BooleanArray`)

Comment (ritchie46) [reply src/compute/boolean.rs:196]: For every bitmap? Ooooh.. then that is definitely a fast path. :)

**Comment by ritchie46** [comment]
Used the cached null counts and added the `all` kernel. Good to go on my part.

**Comment by jorgecarleitao** [comment]
Thanks a lot, @ritchie46 !

**Pull request merged**